executionlayer/src/main/kotlin/maru/executionlayer/manager/JsonRpcExecutionLayerManager.kt

```diff
@@ -223,11 +168,63 @@ class JsonRpcExecutionLayerManager private constructor(
     headHash: ByteArray,
     safeHash: ByteArray,
     finalizedHash: ByteArray,
-  ): SafeFuture<ForkChoiceUpdatedResult> {
-    TODO("Will implement once the chain following feature is implemented")
-  }
+    payloadAttributes: PayloadAttributes?,
+  ): SafeFuture<ForkChoiceUpdatedResult> =
+    executionLayerClient
+      .forkChoiceUpdate(
+        ForkChoiceStateV1(
+          Bytes32.wrap(headHash),
+          Bytes32.wrap(safeHash),
+          Bytes32.wrap(finalizedHash),
+        ),
+        payloadAttributes?.toPayloadAttributesV1(),
+      ).thenCompose { response ->
+        log.debug("Forkchoice update response with payload attributes {}", response)
+        if (response.isFailure) {
+          // TODO: Temporary hack for protocol switches. Should go when QBFT fully works along with dummy consensus
+          executionLayerClient
+            .forkChoiceUpdate(
+              ForkChoiceStateV1(
+                Bytes32.wrap(headHash),
+                Bytes32.wrap(safeHash),
+                Bytes32.wrap(finalizedHash),
+              ),
+              null,
+            )
+        } else {
+          SafeFuture.completedFuture(response)
+        }
+      }.thenApply { response ->
+        log.debug("Forkchoice update response after a retry without payload attributes {}", response)
+        if (response.isFailure) {
+          throw IllegalStateException(
+            "forkChoiceUpdate request failed! nextBlockTimestamp=${
+              payloadAttributes?.timestamp
+            } " + response.errorMessage,
+          )
+        } else {
+          mapForkChoiceUpdatedResultToDomain(response)
+        }
+      }
 
-  override fun importBlock(executionPayload: ExecutionPayload): ByteArray {
-    TODO("Will implement once the chain following feature is implemented")
-  }
+  override fun importBlock(executionPayload: ExecutionPayload): SafeFuture<Unit> =
```

Comment:
Better name for this would be `newPayload` as that is what is called in the engine API. Just doing `newPayload` by itself isn't a complete import we also need to do a FCU.

Reply:
`Block` means something different here. El is importing a block, but that block isn't necessarily the head of the chain. But I'll rename this method for extra clarity.